pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2254,6 +2277,16 @@ private void deleteTableOnServer(String tableNameWithType) {
     }
   }
 
+  public void updateSegmentInterval(String tableNameWithType, SegmentZKMetadata segmentZKMetadata,
+      Schema newSchema) {
+    String segmentName = segmentZKMetadata.getSegmentName();
+    TableConfig tableConfig = getTableConfig(tableNameWithType);
```

Comment:
We don't want to read table config for each segment. We should read the table config once, and pass it into this method